webui/react/src/utils/number.ts

```diff
@@ -2,6 +2,21 @@ import { isString } from './data';
 
 const PERCENT_REGEX = /^(\d+\.?\d*|\.\d+)%$/;
 
+export const findFactorOfNumber = (n: number): number[] => {
+  const abs = Math.abs(n);
+  const factors = [];
+
+  for (let i = 1; i <= Math.floor(Math.sqrt(abs)); i++) {
+    if (abs % i !== 0) continue;
+    factors.push(i);
+
+    if (abs/i === i) continue;
+    factors.push(abs/i);
+  }
+
+  return factors.sort((a, b) => a - b);
```

Comment:
one way to avoid sorting is to just have two sets of factors, one for all the `i`s (aFactors) and one for all the `abs/i`s (bFactors). insert the `aFactors` with push and unshift for `bFactors` then return [ ...aFactors, ...bFactors ]. This is untested but the idea is that the cost of joining two already sorted arrays should be cheaper than iterating through an array and sorting them (best performance being `n log n`)

Reply:
pushed.